pkgs/native_assets_builder/lib/src/model/forge_result.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:collection/collection.dart';
+import 'package:native_assets_cli/native_assets_cli_internal.dart';
+
+import '../../native_assets_builder.dart';
+
+/// The result from a [NativeAssetsBuildRunner.build] or
+/// [NativeAssetsBuildRunner.link].
+final class ForgeResult implements BuildResult, DryRunResult, LinkResult {
```

Comment:
`HookResult`?

(I like forge, but it's confusing because we don't use the name anywhere else. Don't add cognitive load to someone else trying to understand this code in the future.)